.semgrep/semgrep.yml

```
@@ -0,0 +1,153 @@
+        - "domains/**/application/**"
+        - "domains/**/domain/**"
+        - "domains/**/presentation/**"
+
+  # 3) Controller 계층: 허용된 HttpStatus만 사용 (단일 규칙)
+  - id: controller-httpstatus-whitelist
+    languages: [java]
+    message: "허용된 HttpStatus(400/401/403/404/409)만 사용하세요."
+    severity: ERROR
+    pattern: HttpStatus.$S
+    pattern-not: HttpStatus.BAD_REQUEST
+    pattern-not: HttpStatus.UNAUTHORIZED
```

YAMLlint (1.37.1)

[error] duplication of key "pattern-not" in mapping

(key-duplicates)